src/include/function/list/vector_list_functions.h

```diff
@@ -1,220 +1,13 @@
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, int64_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::INT32: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, int32_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::INT16: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, int16_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::INT8: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, int8_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::UINT64: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, uint64_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::UINT32: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, uint32_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::UINT16: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, uint16_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::UINT8: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, uint8_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::INT128: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, common::int128_t,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::DOUBLE: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, double,
-                    OPERATION>;
-        } break;
-        case common::LogicalTypeID::FLOAT: {
-            scalarFunction->execFunc =
-                ScalarFunction::UnaryExecNestedTypeFunction<common::list_entry_t, float, OPERATION>;
-        } break;
-        default: {
-            throw common::BinderException(
-                common::stringFormat("Unsupported inner data type for {}: {}", function->name,
-                    common::LogicalTypeUtils::toString(resultType->getLogicalTypeID())));
-        }
-        }
-        return std::make_unique<FunctionBindData>(resultType->copy());
-    }
-};
-
 struct ListCreationFunction {
+    static constexpr const char* name = "LIST_CREATION";
+
     static function_set getFunctionSet();
     static std::unique_ptr<FunctionBindData> bindFunc(
```

Comment:
You don't have to expose bindFunc, i would recommend defining them as static functions in cpp.